cmd/rpcdaemon/commands/eth_accounts.go

```diff
@@ -62,6 +64,22 @@ func (api *APIImpl) GetTransactionCount(ctx context.Context, address libcommon.A
 	}
 
 	if blockNrOrHash.BlockNumber != nil && *blockNrOrHash.BlockNumber == rpc.PendingBlockNumber {
+		// first lets check for any locks on this account before continuing - but don't wait too long
+		waits := 0
+		for {
+			lock := api.SenderLocks.GetLock(address)
+			if lock > 0 {
+				time.Sleep(2 * time.Millisecond)
+				waits++
+				if waits > 250 {
+					log.Debug("waiting too long for transaction processing, returning default behaviour for nonce", "address", address.String())
```

Comment:
Just curious, does failing to wait for the lock causes some error on client side due to nonce being incorrect ans subsequent error in the next request?

Reply:
It does.  there is a race condition essentially.  If you send a transaction but don't wait to receive the hash for it (signalling that it has been accepted) and make a request for a `pending` nonce then you will potentially get the `latest` from state because the pool hasn't finished processing the TX you just sent in.

The aim of the PR here is to give a little time to allow the pool to finish processing the TX before the nonce is actually checked.

There is a client using cdk-erigon and they rely on using `pending` nonce checks in between transactions and get caught out because of this race condition.